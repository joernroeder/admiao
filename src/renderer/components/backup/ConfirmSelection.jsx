import React, { useContext } from 'react'

import { ThemeContext } from '@emotion/core'
import {
	useSeedPartsDispatch,
	useSeedPartsState,
} from '../../store/SeedPartsStore'

import GridWrap from '../GridWrap'
import Cell from '../Cell'
import SubHeading from '../SubHeading'
import Text from '../Text'
import TextHeading from '../TextHeading'
import Button from '../Button'
import PieChart from '../PieChart'

const ConfirmSelection = props => {
	const dispatch = useSeedPartsDispatch()
	const { uniquePartsN, requiredPartsT } = useSeedPartsState()
	console.log(props)
	const { colors: themeColors } = useContext(ThemeContext)
	const { lightGray = '#f00', black = '#000' } = themeColors

	const onConfirmClick = () => {
		dispatch({ type: 'CONFIRM' })
	}

	const getPieChartData = () => {
		const value = 100 / uniquePartsN

		return Array(uniquePartsN)
			.fill({ value, color: lightGray })
			.map((el, index) => {
				return index < requiredPartsT ? { ...el, color: black } : el
			})
	}

	const pieChartProps = {
		width: 300,
		height: 300,
		radius: 150,
		data: getPieChartData(),
	}

	return (
		<GridWrap>
			<Cell
				gridOffset={2}
				gridColumn={3}
				style={{ textAlign: 'center' }}
				mt={7}
			>
				<PieChart {...pieChartProps} />
				<TextHeading mb={0}>
					{requiredPartsT} of {uniquePartsN}
				</TextHeading>
				<Text>Parts Required</Text>
			</Cell>
			<Cell gridOffset={1} gridColumn={5} mt={9}>
				<SubHeading mb={2}>Confirm Selection</SubHeading>
				<Text mb={9}>
					Donec ullamcorper nulla non metus auctor fringilla. Duis
					mollis, est non commodo luctus, nisi erat porttitor ligula?
				</Text>
				<Button to={'../select-parts'} showArrow={false} mr={3}>
					Nope, go back.
				</Button>
				<Button to={'../how-to-distribute'} onClick={onConfirmClick}>
					Yes, looks good!
				</Button>
			</Cell>
		</GridWrap>
	)
}

export default ConfirmSelection
