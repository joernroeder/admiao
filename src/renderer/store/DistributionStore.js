import React, { createContext, useContext, useReducer, useEffect } from 'react'

const localStorageKey = 'DistributionState'

const initialState = {
	distributionType: undefined,
	distributionIdentifier: undefined,
}

const DistributionTypes = {
	LOCAL_PRINT: 'LOCAL_PRINT',
	REMOTE_PRINT: 'REMOTE_PRINT',
	DIGITAL: 'DIGITAL',
	FILE_SYSTEM: 'FILE_SYSTEM',
}

function reducer(state, action) {
	// sets Distribution Type
	if (Object.keys(DistributionTypes).includes(action.type)) {
		return {
			...state,
			distributionType: DistributionTypes[action.type],
			distributionIdentifier: action.payload || undefined,
		}
	}

	switch (action.type) {
		case 'RESET': {
			localStorage.removeItem(localStorageKey)

			return {
				...initialState,
			}
		}

		case 'SET_IDENTIFIER': {
			return {
				...state,
				distributionIdentifier: action.payload,
			}
		}

		default:
			return state
	}
}

const DistributionStateContext = createContext(undefined)
const DistributionDispatchContext = createContext(undefined)

function DistributionProvider({ children }) {
	const [state, dispatch] = useReducer(
		reducer,
		initialState,
		initialState => {
			const previousState = localStorage.getItem(localStorageKey)

			return previousState ? JSON.parse(previousState) : initialState
		}
	)

	// store distributionType in local storage.
	// This stores the distribution type to improve the backup process.
	useEffect(() => {
		if (state.isInitialState) {
			return
		}

		// filter out empty and initial states
		if (!Object.values(state).filter(Boolean).length) {
			return
		}

		console.log(Object.values(state).filter(Boolean).length)
		localStorage.setItem(localStorageKey, JSON.stringify(state))
	}, [state])

	return (
		<DistributionStateContext.Provider value={state}>
			<DistributionDispatchContext.Provider value={dispatch}>
				{children}
			</DistributionDispatchContext.Provider>
		</DistributionStateContext.Provider>
	)
}

function useDistributionState() {
	const context = useContext(DistributionStateContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionState must be used within a DistributionProvider'
		)
	}
	return context
}

function useDistributionDispatch() {
	const context = useContext(DistributionDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionDispatch must be used within a DistributionProvider'
		)
	}
	return context
}

export {
	DistributionProvider,
	useDistributionState,
	useDistributionDispatch,
	DistributionTypes,
}
