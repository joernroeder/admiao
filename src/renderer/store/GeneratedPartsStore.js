import React, { createContext, useContext, useReducer } from 'react'

const initialState = {
	parts: [],
	confirmed: [],
}

function reducer(state, action) {
	switch (action.type) {
		case 'SET_PARTS': {
			console.log('setting parts in store', action)
			return {
				...state,
				confirmed: Array(action.payload.length).fill(false),
				parts: action.payload,
			}
		}

		case 'CONFIRMED': {
			const updatedConfirmed = {
				confirmed: state.confirmed.map((isConfirmed, index) => {
					if (index === action.payload) {
						return true
					}

					return isConfirmed
				}),
			}

			return {
				...state,
				...updatedConfirmed,
			}
		}

		case 'RESET': {
			return {
				...initialState,
			}
		}

		default:
			return state
	}
}

const GeneratedPartsStateContext = createContext(undefined)
const GeneratedPartsDispatchContext = createContext(undefined)

function GeneratedPartsProvider({ children }) {
	const [state, dispatch] = useReducer(reducer, initialState)

	return (
		<GeneratedPartsStateContext.Provider value={state}>
			<GeneratedPartsDispatchContext.Provider value={dispatch}>
				{children}
			</GeneratedPartsDispatchContext.Provider>
		</GeneratedPartsStateContext.Provider>
	)
}

function useGeneratedParts() {
	const context = useContext(GeneratedPartsStateContext)

	if (context === undefined) {
		throw new Error(
			'useGeneratedParts must be used within a GeneratedPartsProvider'
		)
	}
	return context
}

function useGeneratedPartsDispatch() {
	const context = useContext(GeneratedPartsDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionDispatch must be used within a GeneratedPartsProvider'
		)
	}
	return context
}

export { GeneratedPartsProvider, useGeneratedParts, useGeneratedPartsDispatch }
