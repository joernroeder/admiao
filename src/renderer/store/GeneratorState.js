import React, { createContext, useContext, useState } from 'react'

const STATES = {
	loading: 'LOADING',
	successfullyGenerated: 'SUCCESSFULLY_GENERATED',
	generationFailed: 'GENERATION_FAILED',
	successfullyDistributed: 'SUCCESSFULLY_DISTRIBUTED',
	distributionFailed: 'DISTRIBUTION_FAILED',
}

const GeneratorStateContext = createContext(undefined)

function GeneratorStateProvider({ children }) {
	const [generatorState, setGeneratorState] = useState({
		state: undefined,
		pendingIdentifier: null,
		error: null,
		// additional data returned by the ipc methods may get stored here.
		// todo be more precise/refactor once all distribution types are implemented
		meta: {},
	})

	return (
		<GeneratorStateContext.Provider
			value={[generatorState, setGeneratorState]}
		>
			{children}
		</GeneratorStateContext.Provider>
	)
}

function useGeneratorState() {
	const context = useContext(GeneratorStateContext)

	if (context === undefined) {
		throw new Error(
			'useGeneratedParts must be used within a GeneratedPartsProvider'
		)
	}
	return context
}

export { GeneratorStateProvider, useGeneratorState, STATES }
