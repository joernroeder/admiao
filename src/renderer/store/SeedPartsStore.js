import React, { createContext, useContext, useEffect, useReducer } from 'react'

const localStorageKey = 'SeedPartsState'

const initialState = {
	uniquePartsN: 3,
	requiredPartsT: 2,

	partsPerDisk: 1,
	requiredDisks: 3,

	recoveredFromPreviousSelection: false,
	isInitialState: true,
}

const calcDisks = (uniquePartsN, requiredPartsT) => {
	const partsPerDisk = requiredPartsT - 1
	const requiredDisks = Math.ceil(uniquePartsN / partsPerDisk)

	return {
		partsPerDisk,
		requiredDisks,
	}
}

function reducer(state, action) {
	const { uniquePartsN, requiredPartsT } = state

	const isModifiedState = {
		recoveredFromPreviousSelection: false,
		isInitialState: false,
	}

	switch (action.type) {
		case 'INCREMENT_REQUIRED': {
			const updatedRequiredPartsT = Math.min(
				requiredPartsT + 1,
				uniquePartsN
			)

			return {
				...state,
				requiredPartsT: updatedRequiredPartsT,
				...calcDisks(state.uniquePartsN, updatedRequiredPartsT),
				...isModifiedState,
			}
		}

		case 'DECREMENT_REQUIRED': {
			const updatedRequiredPartsT = Math.max(requiredPartsT - 1, 1)

			return {
				...state,
				requiredPartsT: updatedRequiredPartsT,
				...calcDisks(state.uniquePartsN, updatedRequiredPartsT),
				...isModifiedState,
			}
		}

		case 'INCREMENT_UNIQUE': {
			const updatedUniquePartsN = uniquePartsN + 1

			return {
				...state,
				uniquePartsN: updatedUniquePartsN,
				...calcDisks(updatedUniquePartsN, state.requiredPartsT),
				...isModifiedState,
			}
		}

		case 'DECREMENT_UNIQUE': {
			const updatedUniquePartsN = Math.max(uniquePartsN - 1, 1)
			// also decrement required parts if they would exceed otherwise
			const updatedRequiredPartsT = Math.min(
				updatedUniquePartsN,
				requiredPartsT
			)

			return {
				...state,
				uniquePartsN: updatedUniquePartsN,
				requiredPartsT: updatedRequiredPartsT,
				...calcDisks(updatedUniquePartsN, updatedRequiredPartsT),
				...isModifiedState,
			}
		}

		case 'CONFIRM': {
			return {
				...state,
				...isModifiedState,
			}
		}

		case 'RESET': {
			localStorage.removeItem(localStorageKey)

			return {
				...initialState,
			}
		}

		default:
			return state
	}
}

const SeedPartsStateContext = createContext(undefined)
const SeedPartsDispatchContext = createContext(undefined)

function SeedPartsProvider({ children }) {
	// uses a previous state from local storage if available or uses the initial one defined above.
	const [state, dispatch] = useReducer(
		reducer,
		initialState,
		initialState => {
			const previousState = localStorage.getItem(localStorageKey)

			return previousState
				? {
						...JSON.parse(previousState),
						recoveredFromPreviousSelection: true,
				  }
				: initialState
		}
	)

	// store current state in local storage.
	// This is used to improve the backup process if the user paused to print the
	// seed templates.
	useEffect(() => {
		if (state.isInitialState) {
			return
		}

		localStorage.setItem(localStorageKey, JSON.stringify(state))
	}, [state])

	return (
		<SeedPartsStateContext.Provider value={state}>
			<SeedPartsDispatchContext.Provider value={dispatch}>
				{children}
			</SeedPartsDispatchContext.Provider>
		</SeedPartsStateContext.Provider>
	)
}

function useSeedPartsState() {
	const context = useContext(SeedPartsStateContext)

	if (context === undefined) {
		throw new Error(
			'useSeedPartsState must be used within a SeedPartsProvider'
		)
	}
	return context
}

function useSeedPartsDispatch() {
	const context = useContext(SeedPartsDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useSeedPartsDispatch must be used within a SeedPartsProvider'
		)
	}
	return context
}

export { SeedPartsProvider, useSeedPartsState, useSeedPartsDispatch }
