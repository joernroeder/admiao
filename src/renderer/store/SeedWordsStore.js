import React, { createContext, useContext, useReducer } from 'react'
const bip39 = require('bip39')

const initialWords = [
	/*
	'answer',
	'dice',
	'warm',
	'cup',
	'shock',
	'orbit',
	'pistol',
	'seat',
	'modify',
	'matrix',
	'genre',
	'renew',
	*/
]

const initialState = {
	words: initialWords,
	total: 12, // todo handle other seeds 18, 24
	isConfirmed: false,
	isValid: getIsValid(initialWords),
}

function getIsValid(words) {
	return bip39.validateMnemonic(words.join(' '))
}

function wordsReducer(state, action) {
	const { words } = state

	switch (action.type) {
		case 'APPEND': {
			const updatedWords = [...words, action.payload]

			return {
				...state,
				words: updatedWords,
				isValid: getIsValid(updatedWords),
			}
		}

		case 'RESET': {
			return {
				...state,
				words: [],
				isConfirmed: false,
				isValid: false,
			}
		}

		case 'CLEANUP_SEED': {
			return {
				...state,
				words: [],
			}
		}

		case 'POP': {
			const updatedWords = [...words.slice(0, -1)]

			return {
				...state,
				words: updatedWords,
				isValid: getIsValid(updatedWords),
			}
		}

		case 'CONFIRM': {
			return {
				...state,
				isConfirmed: true,
			}
		}

		default:
			return state
	}
}

const SeedWordsStateContext = createContext(undefined)
const SeedWordsDispatchContext = createContext(undefined)

function SeedWordsProvider({ children }) {
	const [state, dispatch] = useReducer(wordsReducer, initialState)

	return (
		<SeedWordsStateContext.Provider value={state}>
			<SeedWordsDispatchContext.Provider value={dispatch}>
				{children}
			</SeedWordsDispatchContext.Provider>
		</SeedWordsStateContext.Provider>
	)
}

function useSeedWordsState() {
	const context = useContext(SeedWordsStateContext)

	if (context === undefined) {
		throw new Error(
			'useSeedWordsState must be used within a SeedWordsProvider'
		)
	}
	return context
}

function useSeedWordsDispatch() {
	const context = useContext(SeedWordsDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useSeedWordsDispatch must be used within a SeedWordsProvider'
		)
	}
	return context
}

export {
	SeedWordsProvider,
	useSeedWordsState,
	useSeedWordsDispatch,
	wordsReducer,
}
