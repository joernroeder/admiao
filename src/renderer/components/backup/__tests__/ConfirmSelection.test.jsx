import React from 'react'

import { ThemeProvider } from 'emotion-theming'
import { fireEvent, render } from '@testing-library/react'

import ConfirmSelection from '../ConfirmSelection'
import { SeedPartsProvider } from '../../../store/SeedPartsStore'

describe('Confirm Selection', () => {
	const theme = {
		colors: {},
	}

	beforeEach(() => {
		localStorage.clear()
	})

	describe('part selection should be displayed correctly', () => {
		test('correctly display 2 of 3', () => {
			const { getByText, container } = render(
				<ThemeProvider theme={theme}>
					<SeedPartsProvider>
						<ConfirmSelection />
					</SeedPartsProvider>
				</ThemeProvider>
			)

			expect(getByText('2 of 3')).toBeInTheDocument()
			expect(
				container.querySelectorAll('.pieSVG path[fill="#f00"]').length
			).toEqual(2)
			expect(
				container.querySelectorAll('.pieSVG path[fill="#ccc"]').length
			).toEqual(1)

			expect(container.firstChild).toMatchSnapshot()
		})

		test('correctly display 5 of 8', () => {
			// i know that this is an implementation detail which i am manipulating here...
			localStorage.setItem(
				'SeedPartsState',
				'{"uniquePartsN":8,"requiredPartsT":5}'
			)

			const { getByText, container } = render(
				<ThemeProvider theme={theme}>
					<SeedPartsProvider>
						<ConfirmSelection />
					</SeedPartsProvider>
				</ThemeProvider>
			)

			expect(getByText('5 of 8')).toBeInTheDocument()
			expect(
				container.querySelectorAll('.pieSVG path[fill="#f00"]').length
			).toEqual(5)
			expect(
				container.querySelectorAll('.pieSVG path[fill="#ccc"]').length
			).toEqual(3)

			expect(container.firstChild).toMatchSnapshot()
		})
	})

	test('next button should link to /how-to-distribute page', () => {
		const { history, getByText, debug } = global.renderWithRouter(
			<ThemeProvider theme={theme}>
				<SeedPartsProvider>
					<ConfirmSelection />
				</SeedPartsProvider>
			</ThemeProvider>
		)

		// headline
		expect(getByText('Confirm Selection')).toBeInTheDocument()
		expect(getByText('2 of 3')).toBeInTheDocument()

		fireEvent.click(getByText('Yes, looks good!'))

		expect(history.location.pathname).toEqual('/how-to-distribute')
	})

	test('back button should link to /select-parts page', () => {
		localStorage.setItem(
			'SeedPartsState',
			'{"uniquePartsN":8,"requiredPartsT":4}'
		)

		const { history, getByText, debug } = global.renderWithRouter(
			<ThemeProvider theme={theme}>
				<SeedPartsProvider>
					<ConfirmSelection />
				</SeedPartsProvider>
			</ThemeProvider>
		)

		// headline
		expect(getByText('Confirm Selection')).toBeInTheDocument()
		expect(getByText('4 of 8')).toBeInTheDocument()

		fireEvent.click(getByText('Nope, go back.'))

		expect(history.location.pathname).toEqual('/select-parts')
	})
})
