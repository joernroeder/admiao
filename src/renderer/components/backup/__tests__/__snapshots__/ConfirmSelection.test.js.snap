// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Selection part selection should be displayed correctly correctly display 2 of 3 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 256px;
  padding-left: 4px;
  padding-right: 4px;
  width: 25%;
}

.emotion-0 {
  color: black;
  margin-bottom: 0;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  -webkit-letter-spacing: 0.015em;
  -moz-letter-spacing: 0.015em;
  -ms-letter-spacing: 0.015em;
  letter-spacing: 0.015em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-1 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 9px;
  padding-left: 4px;
  padding-right: 4px;
  width: 41.66666666666667%;
}

.emotion-3 {
  color: black;
  margin-bottom: 8px;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

@media screen and (min-width:52em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

.emotion-4 {
  color: black;
  margin-bottom: 9px;
  margin-top: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-5 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: black;
  background-color: transparent;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
}

.emotion-5 .arrow {
  display: inline-block;
  margin-left: 0.5em;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-5:hover .arrow {
  -webkit-transform: translateX(0.30em);
  -ms-transform: translateX(0.30em);
  transform: translateX(0.30em);
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-6 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: white;
  background-color: black;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
}

.emotion-6 .arrow {
  display: inline-block;
  margin-left: 0.5em;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-6:hover .arrow {
  -webkit-transform: translateX(0.30em);
  -ms-transform: translateX(0.30em);
  transform: translateX(0.30em);
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

<div
  class="emotion-8"
>
  <div
    class="emotion-2"
    style="text-align: center;"
  >
    <svg
      class=" pieSVG"
      viewBox="0 0 300 300"
    >
      <path
        d="M150 150 L 150 0 A 150 150 0 0 1 279.9038105676658 225.00000000000003 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 279.9038105676658 225.00000000000003 A 150 150 0 0 1 20.09618943233417 224.99999999999994 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 20.09618943233417 224.99999999999994 A 150 150 0 0 1 150.00000000000009 0 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
    </svg>
    <h3
      class="emotion-0"
      color="black"
      font-size="1"
    >
      2
       of 
      3
    </h3>
    <p
      class="emotion-1"
      color="black"
      font-size="0"
    >
      Parts Required
    </p>
  </div>
  <div
    class="emotion-7"
  >
    <h2
      class="emotion-3"
      color="black"
      font-size="1,2,2"
    >
      Confirm Selection
    </h2>
    <p
      class="emotion-4"
      color="black"
      font-size="0"
    >
      Donec ullamcorper nulla non metus auctor fringilla. Duis mollis, est non commodo luctus, nisi erat porttitor ligula?
    </p>
    <a
      class="emotion-5"
      href="/select-parts"
    >
      Nope, go back.
    </a>
    <a
      class="emotion-6"
      href="/how-to-distribute"
    >
      Yes, looks good!
      <span
        class="arrow"
      >
        â†’
      </span>
    </a>
  </div>
</div>
`;

exports[`Confirm Selection part selection should be displayed correctly correctly display 5 of 8 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 256px;
  padding-left: 4px;
  padding-right: 4px;
  width: 25%;
}

.emotion-0 {
  color: black;
  margin-bottom: 0;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  -webkit-letter-spacing: 0.015em;
  -moz-letter-spacing: 0.015em;
  -ms-letter-spacing: 0.015em;
  letter-spacing: 0.015em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-1 {
  color: black;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 9px;
  padding-left: 4px;
  padding-right: 4px;
  width: 41.66666666666667%;
}

.emotion-3 {
  color: black;
  margin-bottom: 8px;
  margin-top: 0;
  font-size: 14px;
  line-height: 1;
  -webkit-transform: 1px;
  -ms-transform: 1px;
  transform: 1px;
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

@media screen and (min-width:52em) {
  .emotion-3 {
    font-size: 16px;
    line-height: 2;
    -webkit-transform: 2px;
    -ms-transform: 2px;
    transform: 2px;
  }
}

.emotion-4 {
  color: black;
  margin-bottom: 9px;
  margin-top: 0;
  font-size: 12px;
  line-height: 0;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-5 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: black;
  background-color: transparent;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
}

.emotion-5 .arrow {
  display: inline-block;
  margin-left: 0.5em;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-5:hover .arrow {
  -webkit-transform: translateX(0.30em);
  -ms-transform: translateX(0.30em);
  transform: translateX(0.30em);
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-6 {
  border-width: 2px;
  border-color: black;
  border-style: solid;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
  line-height: 0;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 12px;
  line-height: 0;
  -webkit-transform: 0;
  -ms-transform: 0;
  transform: 0;
  color: white;
  background-color: black;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
}

.emotion-6 .arrow {
  display: inline-block;
  margin-left: 0.5em;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-6:hover .arrow {
  -webkit-transform: translateX(0.30em);
  -ms-transform: translateX(0.30em);
  transform: translateX(0.30em);
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

<div
  class="emotion-8"
>
  <div
    class="emotion-2"
    style="text-align: center;"
  >
    <svg
      class=" pieSVG"
      viewBox="0 0 300 300"
    >
      <path
        d="M150 150 L 150 0 A 150 150 0 0 1 256.06601717798213 43.93398282201787 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 256.06601717798213 43.93398282201787 A 150 150 0 0 1 300 150 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 300 150 A 150 150 0 0 1 256.06601717798213 256.06601717798213 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 256.06601717798213 256.06601717798213 A 150 150 0 0 1 150.00000000000003 300 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 150.00000000000003 300 A 150 150 0 0 1 43.93398282201788 256.06601717798213 L 150 150"
        fill="#f00"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 43.93398282201788 256.06601717798213 A 150 150 0 0 1 0 150.00000000000003 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 0 150.00000000000003 A 150 150 0 0 1 43.93398282201785 43.933982822017896 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
      <path
        d="M150 150 L 43.93398282201785 43.933982822017896 A 150 150 0 0 1 149.99999999999997 0 L 150 150"
        fill="#ccc"
        stroke="#fff"
        stroke-width="3"
      />
    </svg>
    <h3
      class="emotion-0"
      color="black"
      font-size="1"
    >
      5
       of 
      8
    </h3>
    <p
      class="emotion-1"
      color="black"
      font-size="0"
    >
      Parts Required
    </p>
  </div>
  <div
    class="emotion-7"
  >
    <h2
      class="emotion-3"
      color="black"
      font-size="1,2,2"
    >
      Confirm Selection
    </h2>
    <p
      class="emotion-4"
      color="black"
      font-size="0"
    >
      Donec ullamcorper nulla non metus auctor fringilla. Duis mollis, est non commodo luctus, nisi erat porttitor ligula?
    </p>
    <a
      class="emotion-5"
      href="/select-parts"
    >
      Nope, go back.
    </a>
    <a
      class="emotion-6"
      href="/how-to-distribute"
    >
      Yes, looks good!
      <span
        class="arrow"
      >
        â†’
      </span>
    </a>
  </div>
</div>
`;
